src/library/scala/collection/ArrayOps.scala

```diff
@@ -1263,13 +1263,18 @@ final class ArrayOps[A](val xs: Array[A]) extends AnyVal {
     *  @param  len    the maximal number of elements to copy.
     *  @tparam B      the type of the elements of the array.
     */
-  def copyToArray[B >: A](dest: Array[B], start: Int, len: Int = Int.MaxValue): dest.type = {
-    Array.copy(xs, 0, dest, start, min(xs.length, min(dest.length-start, len)))
-    dest
+  def copyToArray[B >: A](dest: Array[B], start: Int, len: Int = Int.MaxValue): Int = {
+    val copied = min(xs.length, min(dest.length-start, len))
```

Comment:
The maths are not correct in case the `start` parameter is higher than `dest.length`. Consider the case where `start = dest.length + 1` (and `len` is a positive number):

~~~
copied = min(xs.length, min(dest.length - start, len))
copied = min(xs.length, min(-1, len)
copied = min(xs.length, -1)
copied = -1
~~~